llvm/include/llvm/CAS/CachingOnDiskFileSystem.h

```diff
@@ -110,19 +113,23 @@ class CachingOnDiskFileSystem : public ThreadSafeFileSystem {
   }
 
 protected:
-  CachingOnDiskFileSystem(std::shared_ptr<CASDB> DB);
-  CachingOnDiskFileSystem(CASDB &DB);
+  CachingOnDiskFileSystem(std::shared_ptr<CASDB> DB,
+                          std::shared_ptr<ActionCache> Cache);
+  CachingOnDiskFileSystem(CASDB &DB, ActionCache &Cache);
```

Comment:
Why are we passing the cache into the filesystem? It doesn't seem to be used in the .cpp file except to pass it through the constructor/etc.

Reply:
Well, the dependency scanner, grabs the CAS from FileSystem. Maybe we need a different way to create an ActionCache into dep scanner.